pykechain/models/banner.py

```diff
@@ -66,19 +66,23 @@ def edit(
         >>> banner.edit(text='New text here', url=None)
 
         """
+        active = check_type(is_active, bool, 'is_active')
         update_dict = {
-            'text': check_text(text, 'text') or self.text,
+            'text': check_text(text, 'text'),
             'icon': check_text(icon, 'icon') or str(),
             'active_from': check_datetime(active_from, 'active_from') or check_datetime(
                 self.active_from, 'active_from'),
             'active_until': check_datetime(active_until, 'active_until') or check_datetime(
                 self.active_until, 'active_until'),
-            'is_active': check_type(is_active, bool, 'is_active') or self.is_active,
+            'is_active': empty if active is None else active,
             'url': check_url(url) or str(),
         }
 
         update_dict = clean_empty_values(update_dict=update_dict)
 
+        if "text" not in update_dict or update_dict["text"] is None:
```

Comment:
Maybe it's late, but I don't understand how this `if` clause is not doing exactly the same what the `or` clause in the `update_dict` above was doing

Reply:
It's the same, but `clean_empty_values` removes the field `text` if it is not updated. However, the API currently has `text` as a required field (dont know why) and is therefore put back it.